main.py

```diff
@@ -53,7 +53,26 @@ def create_overlays(ground, ego):
     hud_overlay.add_attribute(game_timing, "time", "Time: ", int)
 
     return debug_overlay, hud_overlay
+    
+def process_events(actions_on_key_pressed, actions_on_key_down):
```

Comment:
In my opinion pressed and down could be misunderstood as the same. My suggestion would be to call the toggling action sth like ..._switch

Reply:
Also found pressed and down not optimal, but found no better alternatives.
When reading out the states and actions of keyboard buttons, it appears as kind of a convention that pressed means the state "it is down" and down means the event "it is pressed down right now". (But in one example i saw it the other way around, so it is indeed misleading.
I went with those names because pygame uses pygame.key.get_pressed() to look for buttons that are in the state where they are pressed down, whereas it uses pygame.locals.KEYDOWN to check for the event where the button gets pressed.
So naming them like this would be not great, but at least in line with the other expressions in pygame.
Swich would be misleading, because it would imply both keydown and keyup events.
Maybe _held_down and _pressed_down? Or _on_pressing and _while_pressed?
I could accept one of those two versions, or leave it as it is.